### ISSUE TITLE
Sorting directories by date

### ISSUE THREAD

**Issue by lknop** [bpatrik/pigallery2#395]
**Describe the bug**

I am not able to get pigallery2 to sort directories by date. Is there anything I am missing? It seems it's using some weird sorting. Is it possible it's some locale/dateformat mixup? Sorting directories by name works as expected.

**Screenshots (optional)**

This is the top part of the list of directories in the gallery root:

```
$ ls --full-time --sort=time
total 784
drwxrwsr-x 1 lknop 100  4096 2021-11-23 19:27:32.000000000 +0100 2021_11
drwxrwsr-x 1 lknop 100  4096 2021-11-04 12:50:52.000000000 +0100 2021_10
drwxrwsr-x 1 lknop 100 12288 2021-09-20 23:30:00.000000000 +0200 2021_09
drwxrwsr-x 1 lknop 100 12288 2021-09-01 01:57:08.000000000 +0200 2021_08
drwxrwsr-x 1 lknop 100  4096 2021-09-01 00:04:55.000000000 +0200 canon_2021_08
drwxrwsr-x 1 lknop 100  4096 2021-09-01 00:01:51.000000000 +0200 canon_2021_07
drwxrwsr-x 1 lknop 100 12288 2021-07-20 23:30:00.000000000 +0200 2021_07
drwxrwsr-x 1 lknop 100   326 2021-07-11 23:49:27.000000000 +0200 canon_2020_08
drwxrwsr-x 1 lknop 100  4096 2021-07-11 23:46:43.000000000 +0200 canon_2020_09
drwxrwsr-x 1 lknop 100   326 2021-07-11 23:45:08.000000000 +0200 canon_2020_11
drwxrwsr-x 1 lknop 100  4096 2021-07-11 23:44:17.000000000 +0200 canon_2020_12
drwxrwsr-x 1 lknop 100   326 2021-07-11 23:43:03.000000000 +0200 canon_2021_01
drwxrwsr-x 1 lknop 100   206 2021-07-11 23:41:58.000000000 +0200 canon_2021_02
drwxrwsr-x 1 lknop 100  8192 2021-07-11 23:39:48.000000000 +0200 canon_2021_06
drwxrwsr-x 1 lknop 100 12288 2021-07-04 01:24:35.000000000 +0200 2021_06
drwxrwsr-x 1 lknop 100  8192 2021-05-31 19:25:19.000000000 +0200 2021_05
drwxrwsr-x 1 lknop 100  8192 2021-04-30 22:49:03.000000000 +0200 2021_04
```

When rendered by pigallery, this is the top of the list when sorted by "descending date". I have enabled "sort directories by time" checkbox. 
![image](https://user-images.githubusercontent.com/12010266/143579701-f664f156-77d3-4d35-8831-903d312eb5ef.png)

**Environment:**
linux amd64 on docker

**Used app version**:
- docker-nightly

**Comment by bpatrik** [comment]
you are probably missing this: 
![kép](https://user-images.githubusercontent.com/7512182/143675781-e71bcdc3-8e96-40e2-aa98-3c57ad5bb36f.png)
Its by default off.

**Comment by bpatrik** [comment]
I mean `sort directories by date`

**Comment by lknop** [comment]
It is enabled, I mentioned it in my report:

![image](https://user-images.githubusercontent.com/12010266/143683760-2bd913b7-d1c4-43bd-867f-28ce0204f63d.png)

I noticed that some of the directories up in the list have a `.thumbnails` folder (created by a different tool) with a recent date. Maybe this is affecting the logic. And today the directory 2021_11 finally landed at the top as I wanted it to, but what caused it to do so is unclear to me.

Can you point me to the place in the code that does directories sorting? Maybe I would be able to discover the cause of the problem.

**Comment by bpatrik** [comment]
Here is the sorting code:

https://github.com/bpatrik/pigallery2/blob/d2d1472763aaa97730f695e1f4ffecb760aab454/src/frontend/app/ui/gallery/gallery.component.ts#L164-L209

here is how the last modified date is calculated: https://github.com/bpatrik/pigallery2/blob/22954550bfd473d85206d62aca50e47853b20862/src/backend/model/threading/DiskMangerWorker.ts#L20-L22

I personally only sort directories based on names. Dir dates is rather flaky. It does not have any reliable metadata. It uses the last modification date that can change based on the OS. Most likely it is based on the last time you touched that dir or any file within that dir.

**Comment by hajo62** [comment]
Is there an option to sort images chronologically - ascending date - AND directories vice versa?
My typically organizing of my photos is having i.e. a vacation in one foldes and name folders with a year and month at the beginning. i.e. _2021.12 - Silvester_ with images like pic01, pic02, pic03...

As you (@bpatrik) wrote, sorting dirs by last modification date is not very reliable. 
And if you have a lot of folders, it's not so handome if the youngst is sorted at the bottom/end which is what you get when sorting photos in a chronological manner

**Comment by bpatrik** [comment]
If you enable directory sorting by date, the app will only sort it by date when you set the overall sorting to date. If you set to name, both directories and photos will be sorted by name.

**Comment by hajo62** [comment]
Thanks for clarifying.
My suggestion or wish was to have an option to sort dirs by name descending - which would sort 2011.12 before 2011.11
AND
images ascending by date - which would sort the welcome drink before the main course.

**Comment by bpatrik** [comment]
Sorry, but I find that confusing. If you show me a well known photo browser / gallery that has this feature, I will consider adding it.

**Comment by hajo62** [comment]
i.e. google photo does _something like that_. 
You can sort albums based on date of included photos which makes the _youngest_ album to appear as first in the list. The single albums can be sorted with oldes first which makes the photos appear in chronological manner.

Or with your `.order_descending_name.pg2conf` _trick_ and your Default photo sorting method set to `ascending date` you can do this with your PiGallery. But you have to put that file into every folder which includes subfolders.

Am I alone find sorting like that - youngest albums first and within an album see the start of the photo series before the end - _natural_? 
<img src="https://user-images.githubusercontent.com/15800980/149639469-289295fd-0945-45f6-93a6-2e765838229d.png" width="50%" height="50%">

**Comment by rcsas** [comment]
Just to say I love PiGallery2, and the only thing I'm missing is the directory sorting like hajo62 proposed - newer directories first, not by directory modification time, but by contained picture EXIF time

**Comment by bpatrik** [comment]
Fixed this in #675. Closing this. Feel free to reopen if something is still missing.
